File: app-miyou/src/main/java/com/miyou/app/domain/credit/port/UserCreditRepository.java
package com.miyou.app.domain.credit.port;

import com.miyou.app.domain.credit.model.UserCredit;
import com.miyou.app.domain.dialogue.model.UserId;
import reactor.core.publisher.Mono;

public interface UserCreditRepository {

	Mono<UserCredit> findByUserId(UserId userId);

	Mono<UserCredit> save(UserCredit userCredit);
}

File: app-miyou/src/main/kotlin/com/miyou/app/domain/credit/port/UserCreditRepository.kt
package com.miyou.app.domain.credit.port

import com.miyou.app.domain.credit.model.UserCredit
import com.miyou.app.domain.dialogue.model.UserId
import reactor.core.publisher.Mono

interface UserCreditRepository {
    fun findByUserId(userId: UserId): Mono<UserCredit>

    fun save(userCredit: UserCredit): Mono<UserCredit>
}
